package/MDAnalysis/visualization/base.py

```diff
@@ -0,0 +1,97 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+import warnings
+
+try:
+    shell = get_ipython()
+except NameError:
+    shell = None
+    warnings.warn("You should be in an interactive python shell (IPython or "
+                  "Notebook) to use this properly")
+
+from .. import _FORMATTERS
+
+
+class _Formattermeta(type):
+    """Automatic Formatter registration metaclass
+
+    .. versionadded:: 2.0.0
+    """
+    # Auto register upon class creation
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+        try:
+            fmt = classdict['format']
+        except KeyError:
+            pass
+        else:
+            _FORMATTERS[fmt] = {}
+
+
+class FormatterBase(metaclass=_Formattermeta):
+    """Base class for formatters
+
+    .. versionadded:: 2.0.0
+    """
+
+    def add_repr(self, obj, mime, func):
+        """Add a new formatter to an object
+
+        Parameters
+        ----------
+        obj : object
+            An MDAnalysis object
```

Comment:
That's a really broad input type! There's no restriction here that it should be related to i.e., `Universe` or `atomgroup` as far as visualization-related things goes?